### PR TITLE
Add isFunded construct to ReleaseGold

### LINKED ISSUE #8486: Support funding of ReleaseGold contracts after deployment
### Expected Behavior

- Deploy ReleaseGold grants from ephemeral keys
- Fund ReleaseGold grants from high-custody key

### Current Behavior

Contracts must be funded at deploy time (in constructor?)

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Removes requirement for funding in `initialize`
- Adds storage boolean `funded`
- Adds view function `isFunded`

### Tested

Unit tests

### Related issues

- Fixes #8486

### Backwards compatibility

With usage of init and accounting for already deployed grants which will be updated

### CODE REVIEW

File: packages/protocol/contracts/governance/ReleaseGold.sol [hunk 1]
```diff
@@ -75,6 +75,9 @@ contract ReleaseGold is UsingRegistry, ReentrancyGuard, IReleaseGold, Initializa
   // Indicates if this schedule's unreleased gold can be used for voting.
   bool public canVote;
 
+  // Indicates if this schedule has received the initial gold for funding.
+  bool public funded = false;
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/ReleaseGold.sol]: Should this be `internal` if `isFunded` has potentially different behavior?

**Review by m-chrzan — COMMENTED** [review]
Some tests covering the new behaviors would be great.

File: packages/protocol/test/governance/voting/release_gold.ts [hunk 2]
```diff
@@ -198,13 +211,78 @@ contract('ReleaseGold', (accounts: string[]) => {
...
+        releaseGoldDefaultSchedule.numReleasePeriods
+      )
+      await goldTokenInstance.transfer(releaseGoldInstance.address, sufficientPrinciple, {
+        from: owner,
+      })
+      const isFunded = await releaseGoldInstance.isFunded()
+      assert.isTrue(isFunded)
+    })
+
+    it('should update isFunded() if schedule principle not fulfilled but has begun releasing', async () => {
```
Comment (m-chrzan) [inline packages/protocol/test/governance/voting/release_gold.ts:272]: Is this actually desired logic? I assume this is just the easiest way to implement this and we're just relying on the human funding the contract to ensure `isFunded` actually becomes true soon after deployment, is that right?

**Review by m-chrzan — APPROVED** [review]
Looks good, just one question. Thanks for the tests!

File: packages/protocol/test/governance/voting/release_gold.ts (hunk 2 above)
Comment (yorhodes) [reply packages/protocol/test/governance/voting/release_gold.ts:272]: I thought it made sense to have client tooling reject if `isFunded` is false but there is no contract behavior that is effected by this status

**Pull request merged**